### PR TITLE
Add reworked configuration docs.

### PR DESCRIPTION
**Pull request by arturdryomov**

### CODE REVIEW

File: docs/configuration/description.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
+Files are placed in the `.mainframer` directory.
+The final configuration is the combination of files placed in:
+
+* `${HOME}/.mainframer` — the global configuration;
```
Comment (artem-zinnatullin) [inline docs/configuration/description.md]: btw, that I would probably like to change to `~/.config/mainframer` 

A lot of other tools store configs there (example from my machine):

```console
tree -L 1 ~/.config
/Users/user/.config
├── alacritty
├── gcloud
├── gtk-2.0
├── htop
├── kitty
├── nvim
└── thefuck
```

Can't find it being described as some standard other than by XDG https://unix.stackexchange.com/questions/68721/where-should-user-configuration-files-go but a lot of tools seem to use it now

File: docs/configuration/description.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+* `.mainframer` — the project configuration.
+
+It is posssible to share the configuration using the global one
```
Comment (artem-zinnatullin) [inline docs/configuration/description.md]: Maybe reword?

"It is possible to declare common configuration using the global one and fine-tune it using the project one"

File: docs/configuration/description.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  name: "{SSH machine name}"
+push:
+  compression: {level}
+pull:
+  compression: {level}
+```
+
+Name               | Optional | Value   | Default | Description
+-------------------|----------|---------|---------|------------------
+`remote.name`      | No       | `string`| —       | Remote machine name from SSH config.
```
Comment (artem-zinnatullin) [inline docs/configuration/description.md]: "Hostname/IP or remote machine name from `$HOME/.ssh/config`."

File: docs/configuration/description.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+```yaml
+remote:
+  name: "{SSH machine name}"
+push:
+  compression: {level}
+pull:
+  compression: {level}
+```
+
+Name               | Optional | Value   | Default | Description
```
Comment (artem-zinnatullin) [inline docs/configuration/description.md]: Usually it's "Required" instead of "Optional" :)

File: docs/configuration/description.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+`pull.compression` | Yes      | `0..9`  | `0`     | Compression level used to copy files from remote machine to local one.
+
+Compression level is inherited from underlying `rsync`
+which uses [`zlib` values](https://www.zlib.net/manual.html):
+
+> The compression level must be between `0` and `9`:
+> `1` gives best speed, `9` gives best compression, `0` gives no compression at all
+> (the input data is simply copied a block at a time).
+
+## `.mainframer/ignore.yml`
```
Comment (artem-zinnatullin) [inline docs/configuration/description.md:46]: This section misses description

File: docs/configuration/description.md (hunk 1 above)
Comment (arturdryomov) [reply docs/configuration/description.md]: Sounds good. Yes, it is XDG-only, but I don’t mind a semi-clean home directory on macOS.

File: docs/configuration/description.md (hunk 5 above)
Comment (arturdryomov) [reply docs/configuration/description.md:46]: Files are described in the `.mainframer` section above.

File: docs/configuration/description.md (hunk 1 above)
Comment (artem-zinnatullin) [reply docs/configuration/description.md]: My example is from macOS just in case :)

File: docs/configuration/description.md (hunk 5 above)
Comment (artem-zinnatullin) [reply docs/configuration/description.md:46]: Yeah, I guess I just wanted to see description there, ok let's go without it for now then

**Comment by arturdryomov** [comment]
@artem-zinnatullin, updated.

File: docs/configuration/description.md [hunk 6]
```diff
@@ -0,0 +1,69 @@
+Files are placed in the `.mainframer` directory.
```
Comment (artem-zinnatullin) [inline docs/configuration/description.md]: This sentence is overlapped with next paragraph and also is incorrect in case of `${HOME}/.config/mainframer`

File: docs/configuration/description.md [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+push:
+  compression: {level}
+pull:
+  compression: {level}
+```
+
+Name               | Required | Value   | Default | Description
+-------------------|----------|---------|---------|------------------
+`remote.name`      | Yes      | `string`| —       | Remote machine name from SSH config or hostname / IP address.
+`push.compression` | No       | `0..9`  | `0`     | Compression level used to copy files from local machine to remote one.
```
Comment (artem-zinnatullin) [inline docs/configuration/description.md:36]: default is 3

File: docs/configuration/description.md [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+`pull.compression` | No       | `0..9`  | `0`     | Compression level used to copy files from remote machine to local one.
```
Comment (artem-zinnatullin) [inline docs/configuration/description.md:37]: default is 1

**Review by artem-zinnatullin — APPROVED** [review]

File: docs/configuration/description.md (hunk 8 above)
Comment (arturdryomov) [reply docs/configuration/description.md:37]: Are these our values? Do you have a reference otherwise? I’ve found [an outdated email](http://samba.2283325.n4.nabble.com/what-is-the-default-value-for-compress-level-td2538905.html). [The source code says otherwise](https://github.com/kuna-systems/rsync/blob/5546dab32970955e77ef7a5886bcd8fb765a25bf/options.c#L1825-L1850) — seems to be `-1` (`Z_DEFAULT_COMPRESSION` value).

Comment (arturdryomov) [reply docs/configuration/description.md:37]: MF `2.x` uses `1` by default for both cases.

Comment (artem-zinnatullin) [reply docs/configuration/description.md:37]: It's in 3.x PR #239, idea is that usually you push text files which are very compressible and pull binaries which are barely compressible, thus push `3` and pull `1`

Comment (artem-zinnatullin) [reply docs/configuration/description.md:37]: We can update this doc in #239, going to merge your PR now
